dash/backend/src/modules/auth/services/oauth2/azure-oauth2.service.ts

```diff
@@ -29,17 +31,25 @@ export class AzureOauth2Service extends Oauth2AuthProvider {
     });
     const userEmail = oAuthUserProfile.data?.userPrincipalName;
     const emailDomain = userEmail.split('@').pop().toLowerCase().trim();
+
     if (!oAuth2Config.allowedDomains.includes(emailDomain)) {
       // Check that the email domain of the OAuth user is within the allowed list before anything else
       throw new ForbiddenException('Access Denied', 'User is not permitted to access this site');
     } else {
       // If user is in an allowed domain, check to see if they already exist, and if so return the existing user
       const users: UserProfileDto[] = await this.userProfileService.loadUserByEmail(userEmail);
+      const expectedAuthority = await this.determineAuthority(accessToken);
+
       if (users && Array.isArray(users) && users.length > 0) {
         if (!users[0].isActive) {
           throw new Error('This user is not active.');
         } else {
-          return users[0];
+          const user = users[0];
+          if (user.authorities?.length !== 1 || user.authorities[0].id !== expectedAuthority) {
+            user.authorities = [{ id: expectedAuthority }];
+            return this.userProfileService.updateUser(user).then(resp => resp[0]);
```

Comment:
Should this be awaiting the promise's resolution?

Reply:
In this particular case, it does not matter.

As it currently is, it will return the promise, which will then be evaluated when this function is awaited.

If we added an await, it would evaluated that promise now, then wrap the result in a new promise that would be evaluated when this function is returned.


The times it would be different is if there was synchronous error handling happening (which in this case there is not).

For example, if X(), throws an error, that error will be caught by the try/catch in example A, but not in B as X will not actually throw its error until this function is awaited elsewhere.

```
// A
try {
  return await X();
} catch(err) {
   ...
}

// B
try {
  return X();
} catch (err) {
   ...
}



```